src/ecrc-frontend/src/components/page/applicationForm/ApplicationForm.js

```diff
@@ -97,52 +100,123 @@ export default function ApplicationForm({
   const [mailingProvinceError, setMailingProvinceError] = useState("");
   const [mailingPostalCode, setMailingPostalCode] = useState("");
   const [mailingPostalCodeError, setMailingPostalCodeError] = useState("");
+  const [toTransition, setToTransition] = useState(false);
 
-  const [provinces, setProvinces] = useState([]);
+  const location = useLocation();
 
   useEffect(() => {
-    if (!isAuthorized() || !isActionPerformed("userConfirmation"))
-      setToHome(true);
+    const urlParam = queryString.parse(location.search);
+    const { code } = urlParam;
 
-    let token = sessionStorage.getItem("jwt");
+    const token = sessionStorage.getItem("jwt");
     const uuid = sessionStorage.getItem("uuid");
 
-    const payload = accessJWTToken(token);
-    token = generateJWTToken({
-      ...payload,
-      authorities: ["Authorized", "ROLE"]
-    });
+    if (code) {
+      Promise.all([
+        axios.get(`/ecrc/protected/login?code=${code}&requestGuid=${uuid}`, {
+          headers: {
+            Authorization: `Bearer ${token}`
+          }
+        }),
+        axios.get(`/ecrc/protected/getProvinceList?requestGuid=${uuid}`, {
+          headers: {
+            Authorization: `Bearer ${token}`
+          }
+        })
+      ])
+        .then(res => {
+          sessionStorage.setItem("jwt", res[0].data);
+
+          if (!isAuthorized()) setToHome(true);
+
+          setProvinces(res[1].data.provinces.province);
+
+          const {
+            userInfo: {
+              birthdate,
+              address: { street_address, locality, region, postal_code },
+              gender,
+              given_name,
+              given_names,
+              family_name,
+              identity_assurance_level
+            }
+          } = accessJWTToken(res[0].data);
+
+          // TODO Check identity assurance level
```

Comment:
why is this still a TODO, if we are doing it

Reply:
Should not be!